src/pages/index.js

```diff
@@ -146,12 +204,75 @@ function handleEscape(event) {
 
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
-  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  cardForm.reset();
-  disableButton(cardModalSubmitBtn, settings);
-  closeModal(cardModal);
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .addCardInfo({ name: cardNameInput.value, link: cardLinkInput.value })
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      cardForm.reset();
+      disableButton(cardModalSubmitBtn, settings);
+      closeModal(cardModal);
+    })
+    .catch(console.err)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editavatarInfo(avatarLinkInput.value)
+    .then((data) => {
+      avatarProfile.src = data.avatar;
+      closeModal(avatarModal);
+    })
+    .catch(console.err)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  const deleteBtn = evt.submitter;
+  setDeleteButtonText(deleteBtn, true);
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      cardsList.removeChild(selectedCard);
+      closeModal(deleteModal);
+    })
+    .catch()
```

Comment:
`.catch()` should be `.catch(console.error)` to handle the errors

Reply:
Completed and thanks!!